tests/DataTransferObjectTest.php

```diff
@@ -27,6 +27,7 @@ public function only_the_type_hinted_type_may_be_passed()
         $this->markTestSucceeded();
 
         $this->expectException(DataTransferObjectError::class);
+        $this->expectExceptionMessageRegExp('/Invalid type: expected `class@anonymous[^:]+::foo` to be of type `string`, instead got value ``, which is boolean/');
```

Comment:
Why the need for regex matching? Wouldn't it be possible to do a string equals string assertion?

Reply:
Unfortunately not, because they contain some PHP internal magic object ID stuff which is "random" as far as the test is concerned.

I wish I wouldn't have to do that 😄 

This only applies to tests using anonymous classes within.

Reply:
_and_ it also includes:
- the local path 💥
- a (I think) `0`-byte 🤷‍♀️
  Pretty dark magic 🧙‍♂️ going on there


Here's an example how the error looks if it doesn't mach:
> `Failed asserting that exception message 'Invalid type: expected class@anonymous/Users/neo/src/data-transfer-object/tests/DataTransferObjectTest.php0x1126bc4d3::foo to be of type string, instead got value , which is boolean.' matches '/Invalid type: expected class@annymous[^:]+::foo to be of type string, instead got value "", which is boolean/'.`

Reply:
I see, thanks for clarifying!